### PR TITLE
Deploy HDInsight cluster in VNet

### PR DESCRIPTION
**Pull request by algattik**
HDInsight cluster was not being deployed in VNet, added necessary config

### CODE REVIEW

File: 101-hdinsight-secure-vnet/azuredeploy.json [hunk 1]
```diff
@@ -47,7 +47,9 @@
       "name": "[concat(parameters('clusterName'),'-vnet')]",
       "addressSpacePrefix": "10.0.0.0/16",
       "subnetName": "subnet1",
-      "subnetPrefix": "10.0.0.0/24"
+      "subnetPrefix": "10.0.0.0/24",
+      "id": "[resourceId('Microsoft.Network/virtualNetworks', concat(parameters('clusterName'),'-vnet'))]",
+      "subnet": "[concat(resourceId('Microsoft.Network/virtualNetworks', concat(parameters('clusterName'),'-vnet')), '/subnets/subnet1')]"
```
Comment (bmoore-msft) [inline 101-hdinsight-secure-vnet/azuredeploy.json]:
```suggestion
      "subnet": "[resourceId('Microsoft.Network/virtualNetworks/subnets', concat(parameters('clusterName'),'-vnet')), 'subnet1')]"
```

And then remove the id above

File: 101-hdinsight-secure-vnet/azuredeploy.json [hunk 2]
```diff
@@ -62,27 +64,27 @@
...
-            "name": "hdirule1",
+            "name": "Allow_HDInsight_Management_Traffic",
             "properties": {
-              "description": null,
+              "description": "Allow traffic from the Azure health and management services to reach HDInsight",
               "protocol": "*",
               "sourcePortRange": "*",
               "destinationPortRange": "443",
-              "sourceAddressPrefix": "168.61.49.99",
+              "sourceAddressPrefixes": ["168.61.49.99", "23.99.5.239", "168.61.48.131", "138.91.141.162"],
```
Comment (bmoore-msft) [inline 101-hdinsight-secure-vnet/azuredeploy.json]: what's the source of the ip addresses here?

File: 101-hdinsight-secure-vnet/azuredeploy.json [hunk 3]
```diff
@@ -94,32 +96,18 @@
...
               "protocol": "*",
               "sourcePortRange": "*",
               "destinationPortRange": "443",
-              "sourceAddressPrefix": "168.61.48.131",
+              "sourceAddressPrefixes": ["23.102.235.122", "52.175.38.134", "13.76.245.160", "13.76.136.249", "104.210.84.115", "13.75.152.195", "13.77.2.56", "13.77.2.94", "191.235.84.104", "191.235.87.113", "52.229.127.96", "52.229.123.172", "52.228.37.66", "52.228.45.222", "42.159.96.170", "139.217.2.219", "42.159.198.178", "42.159.234.157", "40.73.37.141", "40.73.38.172", "52.164.210.96", "13.74.153.132", "52.166.243.90", "52.174.36.244", "51.4.146.68", "51.4.146.80", "51.5.150.132", "51.5.144.101", "52.172.153.209", "52.172.152.49", "104.211.223.67", "104.211.216.210", "13.78.125.90", "13.78.89.60", "40.74.125.69", "138.91.29.150", "52.231.39.142", "52.231.36.209", "52.231.203.16", "52.231.205.214", "51.141.13.110", "51.141.7.20", "51.140.47.39", "51.140.52.16", "13.67.223.215", "40.86.83.253", "13.82.225.233", "40.71.175.99", "157.56.8.38", "157.55.213.99", "52.161.23.15", "52.161.10.167", "13.64.254.98", "23.101.196.19", "52.175.211.210", "52.175.222.222"],
```
Comment (bmoore-msft) [inline 101-hdinsight-secure-vnet/azuredeploy.json]: what's the source of the ip addresses here?

File: 101-hdinsight-secure-vnet/azuredeploy.json (hunk 2 above)
Comment (algattik) [reply 101-hdinsight-secure-vnet/azuredeploy.json]: The IPs in question are listed at https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-extend-hadoop-virtual-network. I tried adding that link to the description, but it is limited to 140 chars :(

File: 101-hdinsight-secure-vnet/azuredeploy.json (hunk 3 above)
Comment (algattik) [reply 101-hdinsight-secure-vnet/azuredeploy.json]: As Above. The IPs in question are listed at https://docs.microsoft.com/en-us/azure/hdinsight/hdinsight-extend-hadoop-virtual-network.

**Pull request merged**